samples/bluetooth/bap_broadcast_assistant/src/main.c

```diff
@@ -536,14 +554,49 @@ static struct bt_bap_broadcast_assistant_cb ba_cbs = {
 
 static struct bt_le_per_adv_sync_cb pa_synced_cb = {
 	.synced = pa_sync_synced_cb,
+	.term = pa_sync_term_cb,
 	.recv = pa_recv,
 };
 
 static void reset(void)
 {
-	printk("\n\nReset...\n\n");
+	int err;
+
+	printk("\n\nResetting...\n\n");
+
+	if (broadcast_sink_conn != NULL) {
+		err = bt_conn_disconnect(broadcast_sink_conn, BT_HCI_ERR_LOCALHOST_TERM_CONN);
+
+		if (err != 0) {
+			printk("bt_conn_disconnect failed with %d\n", err);
+		} else {
+			if (k_sem_take(&sem_sink_disconnected, SEM_TIMEOUT) != 0) {
+				/* This should not happen */
+
+				__ASSERT_NO_MSG(false);
+			}
+		}
+		__ASSERT_NO_MSG(err == 0);
+	}
+
+	/* Ignore return value as scanning may already be stopped */
+	(void)bt_le_scan_stop();
```

Comment:
Nit: May want to check for -EALREADY.

Reply:
I did consider that, but as far as I can tell, the function does not return -EALREADY if not scanning. 

I guess it's related to this part in the BT Core spec:
> Disabling scanning when it is disabled has no effect.